### PR TITLE
Updates to PDHG and SPDHG notebooks 

### PR DESCRIPTION
**Pull request by MargaretDuff**
Added partitioner to SPDHG (from https://github.com/TomographicImaging/CIL-Demos/issues/124), fixed bug in SPDHG (https://github.com/TomographicImaging/CIL-Demos/issues/128) and added info about warm start to PDHG notebooks to note change in default.

Needs to wait until release of TV warmstart into CIL.

### CODE REVIEW

**Comment by MargaretDuff** [comment]
@gfardell or @lauramurgatroyd - can you just double check this before I merge

File: demos/2_Iterative/03_PDHG.ipynb
Comment (lauramurgatroyd) [inline demos/2_Iterative/03_PDHG.ipynb]:
```suggestion
    "- Using the TotalVariation class, from CIL. This solves the TV denoising problem (using the FGP algorithm) in CPU. The default warm_start=True, means that the TV denoising problem is initialised each time with the solution of the previous run of the TV denoising problem. This means we can use as few as 10 iterations of FGP and increases computation speed. \n",
```

File: demos/2_Iterative/04_SPDHG.ipynb
Comment (lauramurgatroyd) [inline demos/2_Iterative/04_SPDHG.ipynb:1]: You need to run nbstripout on this file before committing it, so that the outputs are removed. As it is, the diff is too big for me to see what you changed.

**Review by lauramurgatroyd — CHANGES_REQUESTED** [review]
I have requested some changes, but also, we can't merge until we have the new release of CIL with the new warm_start parameter.

**Comment by MargaretDuff** [comment]
> I have requested some changes, but also, we can't merge until we have the new release of CIL with the new warm_start parameter.

Thanks @lauramurgatroyd - that is an excellent point :)

File: demos/2_Iterative/04_SPDHG.ipynb
Comment (MargaretDuff) [reply demos/2_Iterative/04_SPDHG.ipynb:1]: Thanks @lauramurgatroyd - Think I have done this now. Let's wait until the new CIL release

**Comment by lauramurgatroyd** [comment]
Hi, are we able to merge this now?

**Comment by MargaretDuff** [comment]
I think so, `warm_start` got added 23.1.0 which is what we recommend currently for the notebooks

**Review by lauramurgatroyd — CHANGES_REQUESTED** [review]
Most of these files have their outputs rendered - please remove this before merging

**Comment by MargaretDuff** [comment]
Superseded by #165

**Pull request closed without merging**
